sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1010,7 +1014,16 @@ class Analyzer(
                 s"(valid range is [1, ${aggs.size}])")
           case o => o
         }
-        Aggregate(newGroups, aggs, child)
+        // If an aggregateExpression is integer, after replaced with this aggregateExpression, the
+        // groupExpression will be considered as an ordinal still, so we should filter it.
```

Comment:
Please also comment on why this filtering is safe from changing grouping result. Thanks.